src/capability/numbergen/mod.rs

```diff
@@ -1,47 +1,56 @@
 /// [rand](::rand) crate adaptors for random number generation capability
-#[cfg(feature = "rand")]
 pub mod rand;
 
-#[cfg(feature = "rand")]
 pub use self::rand::Numbergen as RandNumbergen;
 
-pub use uuid;
-pub use uuid::Uuid;
+pub use uuid::{self, Uuid};
 
 use core::fmt::Debug;
 
-use async_trait::async_trait;
-use wascap::jwt;
-
-/// Builtin random number generation capability available within `wasmcloud:builtin:numbergen` namespace
-#[async_trait]
-pub trait Numbergen: Sync + Send {
-    /// Error returned by random number generation operations
-    type Error: ToString + Debug;
+use anyhow::{bail, Context, Error, Result};
+use serde::Serialize;
+use wasmbus_rpc::common::{deserialize, serialize};
+use wasmcloud_interface_numbergen::RangeLimit;
 
+#[derive(Clone, Debug)]
+/// Random number generator invocation
+pub enum Invocation {
     /// Generates a v4 [Uuid]
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Self::Error`] if the operation fails
-    async fn generate_guid(&self, claims: &jwt::Claims<jwt::Actor>) -> Result<Uuid, Self::Error>;
-
+    GenerateGuid,
     /// Returns a random [u32] within inclusive range from `min` to `max`
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Self::Error`] if the operation fails
-    async fn random_in_range(
-        &self,
-        claims: &jwt::Claims<jwt::Actor>,
+    RandomInRange {
+        /// Minimum [u32] to return
         min: u32,
+        /// Maximum [u32] to return
         max: u32,
-    ) -> Result<u32, Self::Error>;
-
+    },
     /// Returns a random [u32]
-    ///
-    /// # Errors
-    ///
-    /// Returns [`Self::Error`] if the operation fails
-    async fn random_32(&self, claims: &jwt::Claims<jwt::Actor>) -> Result<u32, Self::Error>;
+    Random32,
+}
+
+impl<O: AsRef<str>, P: AsRef<[u8]>> TryFrom<(O, Option<P>)> for Invocation {
```

Comment:
this type signature makes my eyes water 🤣

Reply:
I'll move these to `where`

Reply:
i was mostly just kidding but yeah I think it would be easier to read as a where

Reply:
updated

Reply:
https://github.com/wasmCloud/wasmCloud/compare/169f3052cc579f854885a1be09ba89303c630fd7..3ddbe080c8e302d35019fbe5ca3c906c7186f61a